client/landing/gutenboarding/components/mshots-image/index.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import classnames from 'classnames';
+import { Spinner } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface Props {
+	src: string;
+	alt: string;
+}
+
+const MshotsImage: React.FunctionComponent< Props > = ( { src, alt } ) => {
+	const [ resolvedUrl, setResolvedUrl ] = React.useState< string | undefined >();
+	const [ count, setCount ] = React.useState( 0 );
+	const [ isVisible, setIsVisible ] = React.useState( false );
+
+	const mShotsEndpointUrl = src;
+
+	React.useEffect( () => {
+		async function fetchMshots() {
+			const response = await window.fetch( mShotsEndpointUrl, {
+				method: 'GET',
+				mode: 'cors',
+				// cache: 'no-cache',
+			} );
+
+			if ( response.ok && response.headers.get( 'Content-Type' ) === 'image/jpeg' ) {
+				setResolvedUrl( mShotsEndpointUrl );
+			}
+
+			if ( response.status === 307 ) {
```

Comment:
if `status` is something else (e.g. an error 500), we could just keep looping otherwise for a while and then give up by disabling the preview thumbnail?